drivers/overlay/composefs_supported.go

```diff
@@ -0,0 +1,91 @@
+//go:build linux && composefs && cgo
+// +build linux,composefs,cgo
+
+package overlay
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"sync"
+
+	"github.com/containers/storage/pkg/loopback"
+	"golang.org/x/sys/unix"
+)
+
+var (
+	ComposeFsHelperOnce sync.Once
+	ComposeFsHelperPath string
+	ComposeFsHelperErr  error
+)
+
+func getComposeFsHelper() (string, error) {
+	ComposeFsHelperOnce.Do(func() {
+		ComposeFsHelperPath, ComposeFsHelperErr = exec.LookPath("composefs-from-json")
+	})
+	return ComposeFsHelperPath, ComposeFsHelperErr
+}
+
+func composeFsSupported() bool {
+	_, err := getComposeFsHelper()
+	return err == nil
+}
+
+func generateComposeFsBlob(toc []byte, destFile string) error {
+	outFd, err := unix.Openat(unix.AT_FDCWD, destFile, unix.O_WRONLY|unix.O_CREAT|unix.O_TRUNC|unix.O_EXCL, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to open output file: %w", err)
+	}
+
+
+	writerJson, err := getComposeFsHelper()
+	if err != nil {
+		return fmt.Errorf("failed to find composefs-from-json: %w", err)
+	}
+
+	err = func() error {
+		// Use a func to have a scope for the close.  This must be closed before
+		// fsverity can be enabled.
+		defer unix.Close(outFd)
+
+		fd, err := unix.MemfdCreate("json-toc", unix.MFD_ALLOW_SEALING)
+		if err != nil {
+			return fmt.Errorf("failed to create memfd: %w", err)
+		}
+		defer unix.Close(fd)
+
+		if err := unix.Ftruncate(fd, int64(len(toc))); err != nil {
+			return fmt.Errorf("failed to truncate memfd: %w", err)
+		}
+
+		buf := toc
+		for len(buf) > 0 {
+			n, err := unix.Write(fd, buf)
+			if err != nil {
+				return fmt.Errorf("failed to write to memfd: %w", err)
+			}
+			buf = buf[n:]
+		}
+
+		cmd := exec.Command(writerJson, "--format=erofs", fmt.Sprintf("--out=/proc/self/fd/%d", outFd), fmt.Sprintf("/proc/self/fd/%d", fd))
+		cmd.Stderr = os.Stderr
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("failed to convert json to erofs")
+		}
+		return nil
+	}()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func mountErofsBlob(blobFile, mountPoint string) error {
```

Comment:
Any reason not to use `mount.composefs`?

Reply:
That would be nice. It has some optimizations, like mounting with noacl (which is faster) when there are not acl:s in the erofs image. (This info is in the composefs header at the start of the image file)

Reply:
`mount.composefs` expects a `basedir` option, which I am not using at this moment since this will be picked later when doing the overlay mount.  So I really need to just mount the EROFS file system here